### ISSUE TITLE
[BUG/ISSUE] Boxheight Missing in HEMCO state

### ISSUE THREAD

**Issue by dpfinch** [geoschem/HEMCO#68]
Hi GEOS-Chem Team,

I'm trying to run the HEMCO stand alone (following along with the YouTube tutorial) although at 2x2.5 resolution and using a modified HEMCO_Config.rc file  and the model crashes with this error in the output:

`HEMCO_STANDALONE EXITED WITH ERROR!`

and the following messages in the HEMCO.log file:
```
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
ERROR LOCATION: GetIdx (hco_calc_mod.F90)

ERROR LOCATION: GetIdx (hco_calc_mod.F90)

!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
ERROR LOCATION: GetIdx (hco_calc_mod.F90)

!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
HEMCO ERROR: Boxheight missing in HEMCO state
ERROR LOCATION: GetIdx (hco_calc_mod.F90)

!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
HEMCO ERROR: Boxheight missing in HEMCO state
ERROR LOCATION: GetIdx (hco_calc_mod.F90)
ERROR LOCATION: GetIdx (hco_calc_mod.F90)

!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
HEMCO ERROR: Boxheight missing in HEMCO state

!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
ERROR LOCATION: GetIdx (hco_calc_mod.F90)

!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
HEMCO ERROR:  Error getting vertical index at location           32          87: GFAS_CH4
ERROR LOCATION: GET_CURRENT_EMISSIONS
ERROR LOCATION: HCO_CalcEmis (HCO_CALC_MOD.F90)
ERROR LOCATION: HCO_RUN (hco_driver_mod.F90)

!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
HEMCO ERROR: Error encountered in routine "Hco_Run", phase 2!
ERROR LOCATION:  -> at HCOI_SA_Run (in module HEMCO/Standalone/hcoi_standalone_mod.F90)

!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
HEMCO ERROR: Error encountered in routine "HCOI_Sa_Run"!
ERROR LOCATION:  -> at HCOI_StandAlone_Run (in module HEMCO/Interfaces/hcoi_standalone_mod.F90)
```

Do you know what might be happening and how I can fix it?
Thanks!
Doug

**Comment by msulprizio** [comment]
Thanks for letting us know. I will try to reproduce this on my end and will follow up with suggestions.

**Comment by msulprizio** [comment]
I was not able to reproduce this error using the HEMCO configuration file that ships with GEOS-Chem 13.0.0. Could you upload the modified HEMCO configuration file that you used? There may be a typo or some other issue int hat file that is causing your error.

**Comment by dpfinch** [comment]
I've uploaded the modified HEMCO file (as a .txt to allow upload to GitHub).
I'm using CAMS emissions so have turned off most other sources. Running the stand alone is all in aid of trying to find why my model run has odd isoprene levels with the new emissions so any issues you spot would be very useful!

[HEMCO_Config.rc.txt](https://github.com/geoschem/HEMCO/files/5797189/HEMCO_Config.rc.txt)

**Comment by msulprizio** [comment]
Thanks! I was able to reproduce the error with your HEMCO configuration file. I was able to get around the issue by setting GFAS to false. The issue arises because the GFAS entries utilize the vertical distribution feature of HEMCO and specify `xyL=1:scal300`. When specifying a distribution height HEMCO requires fields like BXHEIGHT be defined. I am still looking into why BXHEIGHT is not getting calculated for the HEMCO standalone and will follow up soon.

**Comment by dpfinch** [comment]
Ah thats great - thanks for the answer! Luckily I don't need GFAS emissions for the time being so I can get on with the other stuff for now!

**Comment by msulprizio** [comment]
This is now fixed in bb42907efe5ae8457ccb9094f292d05a2fa4c361 and will be included in the final HEMCO 3.0.0 / GEOS-Chem 13.0.0 release later this month. To resolve the issue, I needed to move the call to run HEMCO phase 2 (i.e. the calculation of emissions) to after the calls to set and update ExtState within `HEMCO/src/Interfaces/Standalone/hcoi_standalone_mod.F90`.
